src/wallet/wallet.cpp

```diff
@@ -3138,6 +3143,27 @@ int64_t CWallet::GetOldestKeyPoolTime()
     return oldestKey;
 }
 
+void CWallet::MarkDestinationTransactionsDirty(const CTxDestination target_dst)
+{
+    LOCK(cs_wallet);
+
+    for (auto& entry : mapWallet) {
+        CWalletTx& wtx = entry.second;
+
+        for (unsigned int i = 0; i < wtx.tx->vout.size(); i++)
+        {
+            CTxDestination dst;
+
+            if (!ExtractDestination(wtx.tx->vout[i].scriptPubKey, dst))
+                continue;
+
+            if (target_dst == dst) {
+                wtx.MarkDirty();
```

Comment:
you can break here.

Reply:
Won't that skip 2nd+ reuse?

Reply:
The entire transaction will have its balances recomputed, so it doesn't matter if another output in the same tx reuses.

Reply:
done